src/EventSubscriber/CacheTagInvalidatorSubscriber.php

```diff
@@ -83,6 +89,20 @@ public function onReceive(PubSubMessage $message) : void {
       throw new \LogicException('CacheTagsInvalidatorInterface::resetCheckSums() does not exist anymore.');
     }
     $this->cacheTagsInvalidator->resetChecksums();
+
+    // The Purge Queue service temporarily stores all incoming revalidations
+    // in a buffer (refer to: TxBufferInterface) before inserting them into the
+    // queue's storage/database.
+    // This method executed within a long-running loop inside a drush
+    // command that also contains a blocking function.
+    // In certain cases, the buffer is processed only after the loop has ended,
+    // and in some failure scenarios, the buffer is not processed at all
+    // resulting in tags not being purged from varnish.
+    // To ensure the tag revalidations are committed to the queue's database,
+    // the commit method must be invoked to process and finalize the buffer.
+    // @see \Drupal\purge\Plugin\Purge\Queue\QueueService::commit().
+    // @see \Drupal\helfi_proxy\Event\PurgeCommitEvent
+    $this->eventDispatcher->dispatch(new CacheTagInvalidateEvent(), CacheTagInvalidateEvent::EVENT_NAME);
```

Comment:
Event name is optional. We can probably omit it since CacheTagInvalidateEvent() class is not used anywhere else

Reply:
See comment here: https://github.com/City-of-Helsinki/drupal-module-helfi-proxy/pull/79#discussion_r1987077184